tests/integration/test_ingress.py

```diff
@@ -6,29 +6,79 @@
 
 # pylint: disable=unused-argument
 
+import typing
+
 import pytest
 import requests
 from juju.application import Application
 from juju.model import Model
 
 import jenkins
+import state
+from charm import AGENT_DISCOVERY_INGRESS_RELATION_NAME
 
 
 @pytest.mark.abort_on_fail
 async def test_ingress_integration(
-    model: Model, application: Application, ingress_related: Application, external_hostname: str
+    model: Model,
+    application: Application,
+    traefik_application_and_unit_ip: typing.Tuple[Application, str],
+    external_hostname: str,
 ):
     """
     arrange: deploy the Jenkins charm and establish relations via ingress.
     act: send a request to the ingress in /.
     assert: the response succeeds.
     """
-    status = await model.get_status(filters=[application.name])
-    unit = next(iter(status.applications[application.name].units))
-    address = status["applications"][application.name]["units"][unit]["address"]
+    traefik_application, traefik_address = traefik_application_and_unit_ip
+    await application.relate("ingress", traefik_application.name)
+    await model.wait_for_idle(
+        apps=[application.name, traefik_application.name], wait_for_active=True, timeout=20 * 60
+    )
     response = requests.get(
-        f"http://{address}:{jenkins.WEB_PORT}{jenkins.LOGIN_PATH}",
+        f"http://{traefik_address}/{jenkins.LOGIN_URL}",
         headers={"Host": f"{model.name}-{application.name}.{external_hostname}"},
         timeout=5,
     )
-    assert response.status_code == 200
+
+    assert "Authentication required" in str(response.content)
+
+
+# This will only work on microk8s !!
```

Comment:
Is there a way to check this and fail the test if not on microk8s?

Reply:
This test is the test I added for the new integration. That's a good point as well, I'll see if I can add a check and output a meaningful error message when not running on microk8s